### PR TITLE
Import functions directly from the NumPy FFT interface

### PR DESCRIPTION
**Pull request by jakirkham**
Instead of importing the NumPy FFT module interface, import functions from that interface and then wrap those functions. This avoids things leaking in from the NumPy FFT interface into the Dask FFT interface.

### CODE REVIEW

**Comment by jakirkham** [comment]
Thoughts on this, @hgomersall?

**Comment by hgomersall** [comment]
Sorry, I've been really busy this last week.

What's the point of this? I'm not a big fan of overwriting the namespace like you do. What was wrong with the previous way?

**Comment by jakirkham** [comment]
Because we shouldn't have `numpy_fft` available in `dask_fft`.

**Comment by jakirkham** [comment]
> I'm not a big fan of overwriting the namespace like you do.

Also I don't follow this. How are we overwriting the namespace? We are only importing select functions from `numpy_fft` instead of importing the whole module. What is wrong with doing it this way?

File: pyfftw/interfaces/dask_fft.py [hunk 1]
```diff
@@ -53,14 +53,21 @@
 which this may not be true.
 '''
 
-from . import numpy_fft
+from .numpy_fft import (
+    fft,
```
Comment (hgomersall) [inline pyfftw/interfaces/dask_fft.py]: Here `dask_fft.fft` is the same as `numpy_fft.fft`

File: pyfftw/interfaces/dask_fft.py [hunk 2]
```diff
@@ -53,14 +53,21 @@
...
     fft_wrap,
 )
 
-fft = fft_wrap(numpy_fft.fft)
-ifft = fft_wrap(numpy_fft.ifft)
-rfft = fft_wrap(numpy_fft.rfft)
-irfft = fft_wrap(numpy_fft.irfft)
-hfft = fft_wrap(numpy_fft.hfft)
-ihfft = fft_wrap(numpy_fft.ihfft)
+fft = fft_wrap(fft)
```
Comment (hgomersall) [inline pyfftw/interfaces/dask_fft.py]: Here it changes.

**Comment by jakirkham** [comment]
Sure, that is intentional. This is not in any way different from how Python decorators work.

Though maybe I'm not understanding your concern. Could you please clarify what problem you see this causing?

**Comment by hgomersall** [comment]
It's a bit hypothetical, but imported code could change on the fly (particularly if we consider threads). It's not quite how decorators work - they don't put anything in the namespace until after the decorator returns (or rather, the object put into the namespace always has the correct getter associated with it).

**Comment by hgomersall** [comment]
Actually, I don't know how what I said could happen. It's probably safe :) I'm not keen on the style though because it's not clear that the import fft is different to the one in the ultimate module namespace.

**Comment by hgomersall** [comment]
I'd prefer it I think if you kept the existing implementation. You could "protect" `numpy_fft` by importing it with an underscore: `from . import numpy_fft as _numpy_fft`. That would mean it wouldn't import with `import *` nor come up with tab completion in ipython.

**Comment by hgomersall** [comment]
*poke* (cheekily, given by slow response time).

**Comment by hgomersall** [comment]
Great, thanks! I'll merge after successful tests.

**Comment by jakirkham** [comment]
Thanks @hgomersall. Wasn't going to ping you until it passed, but I guess it just did.

**Pull request merged**